### ISSUE TITLE
Additional API endpoint

### ISSUE THREAD

**Issue by jzakotnik** [openkfw/TruBudget#1965]
To better understand, how often data in a TruBudget system is updated, it would be nice to have an API endpoint, that returns the last e.g. 10 updated timestamps, when something was changed in the system. Potentially also if it was project, subproject or workflow item. 

For permission reasons, there doesn't need to be more detailed data.

With the API endpoint, one could integrate this in the dashboard for trubudget operations.

**Comment by amha-mersha** [comment]
Yes, that would be great if it's not a burden.   If you would show me the file/folder I'll be working on some description resources about the project

**Comment by jzakotnik** [comment]
@amha-mersha I believe I would follow this approach:

- Install docker
- Start TruBudget from a container, using 
```
cp scripts/operation/.env.example scripts/operation/.env
bash scripts/operation/start-trubudget.sh --slim
```
- Once started, the API documentation is available under `http://localhost:3000/api/documentation/static/index.html`. So you should see the swagger documentation. Does this work?
- Have a look at `api/src/system/getVersion.ts`, this is a good example which corresponds to `http://localhost:3000/api/documentation/static/index.html#/system/get_api_version`. Here you see the object `multichainClient`, which is the entry point to retrieve respective information from the multichain-blockchain. Does this work? 

Let's have a quick break here and check if this works for you, then we can further deep dive into the code.

**Comment by MartinJurcoGlina** [comment]
@amha-mersha @jzakotnik for development purposes I would refer to these guides that further specify prerequisities, installation etc..
https://github.com/openkfw/TruBudget/blob/main/docs/developer/developer-setup.md 

and running development setup guide here:
https://github.com/openkfw/TruBudget/blob/main/scripts/development/README.md

**Comment by amha-mersha** [comment]
Great I will take it. Thank you

**Comment by jzakotnik** [comment]
@amha-mersha any updates on this?

**Comment by amha-mersha** [comment]
Hello @jzakotnik, I apologize for not finishing this up on time. I've had some classes and have been tight on schedule. If it's important, I'm fine with someone else taking it.

**Comment by amha-mersha** [comment]
Sorry for being late. I set up my development environment following the steps in `https://github.com/openkfw/TruBudget/blob/main/docs/developer/developer-setup.md` and started looking into the project. Running the containers in `trubudget-operation`, especially `trubudget-operation-alpha-api-1` gave me an error saying `Error: Config validation error: "ORGANIZATION_VAULT_SECRET" is not allowed to be empty` but when I checked, it exists in the `.env` file. I can run the containers inside `trubudget-dev`. I am running `trubudget-dev-alpha-api-1` and `trubudget-dev-frontend-1` and I can access `http://localhost:8080/api/documentation/static/index.html#/system/get_api_version`. Hopefully, I am on the right track.

I can go to `http://localhost:8080/api/documentation/static/index.html#/system/get_api_version` and see the `multichainClient` object. How should I proceed from this.
thank you

**Comment by galethil** [comment]
hi @amha-mersha on current main branch you should not be getting error `Error: Config validation error: "ORGANIZATION_VAULT_SECRET"` unless you edit the `scripts/development/.env` file. Do you still have issues or have you been able to start the development environment?

**Comment by amha-mersha** [comment]
hello @galethil , yes I am still getting the issue. I tried to pull the whole thing and run it as @jzakotnik suggested above and containers under `trubudget-operation` start to run except `alpha-api-1` and when I try to run it manually on docker desktop , I get the following error in the terminal
```alpha-api-1   | 
alpha-api-1   | > trubudget-api@2.16.0 start
alpha-api-1   | > node dist/
alpha-api-1   | 
alpha-api-1   | /home/node/dist/config.js:7
alpha-api-1   |     throw new Error(`Config validation error: ${error.message}`);
alpha-api-1   |     ^
alpha-api-1   | 
alpha-api-1   | Error: Config validation error: "ORGANIZATION_VAULT_SECRET" is not allowed to be empty
```
I can't access the page `http://localhost:8080/api/documentation/static/index.html#/system/get_api_version` now

**Comment by amha-mersha** [comment]
I think the `.env.example` inside the `scripts/operation` misses some keys like the `ORGANIZATION_VAULT_SECRET` doesn't have value inside and I tired adding some value to it but I again miss some other values like the `JWT_SECRET` and others. The front end containers work for the `operation` but the `alpha-api-1` work for the `development` for, running the start script inside the `scripts/development` works and I can access  `http://localhost:8080/api/documentation/static/index.html#/system/get_api_version`

**Comment by galethil** [comment]
@amha-mersha  The `scripts/operation` directory is used to run TruBudget in production. Some secret variables are intentionally omitted from the `.env.example` file to ensure that administrators create new custom secret values. Providing a fully populated example might lead administrators to skip creating custom values, which could pose a security risk. This omission is not an issue for `scripts/development` since it is used solely for application development and not in production.

**Comment by amha-mersha** [comment]
I tried to come up with an approach. Simillar the `getVersion` function it will take a `MultichainClient` instance as an argument and from that I can use it's method `listBlocksByHeight` to get a list of block metadatas' (about 10 as specified above). from that extract the timestamp only and return it as a list.
I don't know how I can access the `StreamKind` and also don't know what parameters to pass to the `listBlocksByHeight` method
Am I on the right track?

**Comment by MartinJurcoGlina** [comment]
@amha-mersha from a brief look into it the `getLastBlockInfo` method by the way to get latest block, its info should contain block height that might go as input  into `listBlocksByHeight` selecting latest - 10 to get their info. All interfaces for these methods are defined in `Client.h.ts`

**Comment by amha-mersha** [comment]
Hello, I tried to add the endpoint. A correction/suggestion would be nice.
